server/leafnode.go

```diff
@@ -628,7 +628,12 @@ func (s *Server) createLeafNode(conn net.Conn, remote *leafNodeCfg) *client {
 
 		c.mu.Unlock()
 		// Error will be handled below, so ignore here.
-		c.parse([]byte(info))
+		err = c.parse([]byte(info))
+		if err != nil {
+			c.Debugf("Error reading remote leafnode's INFO: %s", err)
+			c.closeConnection(ReadError)
```

Comment:
I would check for the error returned by parse and just deal with the wrong port error. I would then use `WrongPort` closed reason which may be more appropriate.

Reply:
Thanks, ok so only close connection when the error is wrong port reason and fall through for other errors for now?

Reply:
Yes, so I would have done something like:
```
if err := c.parse([]byte(info)); err == ErrLeafConnectedToWrongPort {
    c.Error(err.Error())
    c.closeConnection(wrongPort)
    return nil
}
```